### ISSUE TITLE
Make more information of gloss publicly accessible

### ISSUE THREAD

**Issue by rem0g** [Signbank/Global-signbank#1403]
For example https://signbank.cls.ru.nl/dictionary/gloss/729 only show video and senses, it's too short of information for non logged in users.

The page should display relation with other glosses, such as variants (METRO-A and METRO-B). 
Phonology
Voorbeeldzinnen
NMM videos
And finally 3-perspective video

@uklomp let me know if you agree with this.

**Comment by uklomp** [comment]
There is indeed a difference between the public version and the gloss-detail page (last one only visible for logged in users). As I understand it, you don't want this difference anymore Gomer? I'd like to know why this difference was there in the first place, maybe there was a good reason for this? @ocrasborn

**Comment by susanodd** [comment]
It's fairly easy to add the requested information.

Since the non-NGT datasets have their own permissions, and since each gloss can be visible or not (In Web setting), then the information that's shown can be controlled that way.

The new Synset Wordnet information might be nice in the public view. There are also social media links in the public view.

**Comment by rem0g** [comment]
> I will discuss this together with Floris and Marloes - on the one hand reconsidering what should be publicly available, on the other hand also keeping in mind that colleagues might be developing a more public friendly interface anyway.

Not to start a discussion here; but since Signbank is a scientifically lexicographically dictionary subsided by the dutch government and they are all available on the Signbank commercially-free; it would not make sense not to publish all relevant details if you get my gist?  

I also want to know opinion from @ocrasborn

**Comment by uklomp** [comment]
> > I will discuss this together with Floris and Marloes - on the one hand reconsidering what should be publicly available, on the other hand also keeping in mind that colleagues might be developing a more public friendly interface anyway.
> 
> Not to start a discussion here; but since Signbank is a scientifically lexicographically dictionary subsided by the dutch government and they are all available on the Signbank commercially-free; it would not make sense not to publish all relevant details if you get my gist?
> 
> I also want to know opinion from @ocrasborn

I completely agree about the subsidy etc.. but at the same time I'm thinking: having an account is not really difficult right? You only need an email address. People get access automatically so I dare say it's quite publicly available. And some might say that the simple overview is actually more user-friendly. 

BUT I also see that we agree that at least for NGT, most (all?) details from the 'details' page could be made publicly available. The question is now if we should consult the other dataset managers first, to implement this for the whole website, and if we want the 'public view' and 'details' to be more alike, or if one of the two pages is superfluous now... also, since we will build another platform together with TYD, should we focus on that instead? These are the questions I want to discuss with Marloes and Floris but feel free to write your opinion here!

**Comment by susanodd** [comment]
The primary difference is the EDIT capability on the Gloss Detail. There is a huge amount of code related to this. The Gloss Edit page has 1000 lines of code just for the Senses, to be able to edit them.

It was decided not to generate (make visible in the browser) all the "edit" related code for the anonymous public page.

**Comment by susanodd** [comment]
I implemented this. There is a pull request. Because of some details with permissions and new code for serving the new kinds of filenames for the NME and perspective videos to anonymous users, it would be most helpful to have some actual videos on the production servers in order to see things.

**Comment by susanodd** [comment]
Screenshots (two different glosses, showing appropriate information being displayed)

![public-view-GEK-A-NGT-top](https://github.com/user-attachments/assets/1c264a04-9d0a-4fe7-aeb3-4c710d3cd1b3)

![public-view-GEK-A-NGT-phonology-relations](https://github.com/user-attachments/assets/991349eb-5563-4ff8-bb37-f7e8176b4ce5)

![public-view-buffelmozarella-tstMH-perspective](https://github.com/user-attachments/assets/795e3d6b-6bbc-4322-b8f3-ea1e71686822)

![public-view-buffelmozarella-tstMH-NME](https://github.com/user-attachments/assets/57d0afc4-655c-433e-93a4-f2ba9dfd63d8)

![public-view-buffelmozarella-tstMH-sentence](https://github.com/user-attachments/assets/430916e6-cbf0-41f7-89cf-58c94349bc93)

**Comment by uklomp** [comment]
Questions to discuss on Thursday: 
1. would it be possible to make changes for public / detailed view for only specific datasets? 
2. would it be easiest to have the exact same lay-out for the things that would be visible on both pages? 
3. how much time would it cost to implement it in the way Gomer proposed? or actually: is this feasible to ask in the upcoming months or is it simply too much?

**Comment by susanodd** [comment]
(1) yes (can add settings fields to Dataset class as is done for field choice hiding)
(2) yes, this is straightforward (I started with the existing layout, so can just adjust the html blocks)
(3) I already implemented what Gomer proposed (but layout needs to follow 2)

**Comment by susanodd** [comment]
@uklomp I revised the screenshot above.

**Comment by uklomp** [comment]
Okay, our conclusion is that if it's indeed "that easy" we would like to have the information that Gomer proposed visible in public view as well, but only for the NGT dataset and only for signs that are "in the dictionary". In the screenshot above, I noticed the following things:
- We'd like the morphology panel there as well
- The panel of the NME video's has a different name here than on the detailed gloss page; it should be the same
- The panels of "relaties met andere gebaren" and "aantekeningen" should not be there

**Comment by susanodd** [comment]
@uklomp One layout question, for "empty" panels, do you want the panel title to show? Or for it to be hidden?

(For display consistency)

On the detail view, there is a tiny glyphicon on top of the panel heading if there is data. 

(The glyphicons may not be consistent? New kinds of data didn't add this. We can also entirely hide the panel heading if there's no data. Depending on the settings for the different public data, then we would hide (not generate) the data.

**Comment by susanodd** [comment]
Semi-related to Search by Morphology

Should the search fields be altered to make them more accessible?

- There are minimal results if you search by "is a blend" or "is part of a blend"


- For sequential morphology, there are the "component N" choices. This is kind of ineffective. This could be changed to "has sequential morphology"

- For the morpheme type, this assumes the Morpheme objects have been used. There are virtually no glosses that use them.

For a researcher, it is handy since if there are glosses the user is aware of, they will know if morphology is missing.

But for an anonymous user, it kind of looks like the feature might be broken if there are no results. (Or at least no public results)

**Comment by susanodd** [comment]
RELATIONS

For variants, the way this is implemented, there is a difference between "pattern matched" versus "relation to other objects".

It was implemented in the past that the user should specify if the objects are related.

(For example, METRO-A, METRO-B.)

**Comment by susanodd** [comment]
> Okay, our conclusion is that if it's indeed "that easy" we would like to have the information that Gomer proposed visible in public view as well, but only for the NGT dataset and only for signs that are "in the dictionary". In the screenshot above, I noticed the following things:
> 
> * We'd like the morphology panel there as well
> * The panel of the NME video's has a different name here than on the detailed gloss page; it should be the same
> * The panels of "relaties met andere gebaren" and "aantekeningen" should not be there

The "relations with other signs" is the "METRO-A" and "METRO-B".

"Aantekeningen" will disappear.

The NME Videos translation is because I'm running it on PyCharm locally, with an outdated translation file. I'll fix that when I get it working on the development server.

**Comment by uklomp** [comment]
> LAYOUT PUBLIC VIEW PANELS
> 
> @uklomp One layout question, for "empty" panels, do you want the panel title to show? Or for it to be hidden?
> 
> (For display consistency)

We want this the same as for the detailed view, which is that the panel title is always visible, right?

**Comment by susanodd** [comment]
This has been updated on `signbank-susan`

All the publicly viewable panels are shown, even if empty.

The sentences are NOT visible now.

Scrolling around as an anonymous user works.

There are hardly any videos on the server.

**Comment by uklomp** [comment]
I checked it and I'm happy that this is the case now. Two more improvements:
1. We are ready to have the panel with example sentences there as well
2. If a panel is empty, could you please make sure the same happens as in the detailed view? In the detailed view, the panel opens although it's empty. In the public view, nothing happens if you click an empty panel so there's no 'feedback' that it's empty (you might think you didn't click it well or that it's broken.

**Comment by susanodd** [comment]
Okay, for (2) I need to adjust where the nesting is on the html elements inside. (At the moment, it checks if it has data. There really is no data. I can make it so there's more empty space.)

**Comment by susanodd** [comment]
I assigned this to myself.

But I need this pull request (code improvements) to be approved since it only makes sense to add more functionality to the improved code (otherwise that work would conflict) #1445 #1454

**Comment by susanodd** [comment]
@uklomp this is on signbank-dev now.

https://signbank-dev.cls.ru.nl/dictionary/gloss/45.html

You can browse without logging in.

It's the database as per this morning.

For some reason, it shows the annotated video, although I don't think it's on the site.

(The path that retrieves it is non-specific to the server. It only shows the /dictionary/protected_media part of the path, not the domain. Perhaps because I have multiple signbanks open on different servers.)

(The code for the annotated sentences is the "display" read-only part that is shown in the Gloss Detail view.)
